### PR TITLE
Enabling Ruff as code linter

### PR DESCRIPTION
**Pull request by cbjuan**
### Summary
This PR enables ruff https://github.com/astral-sh/ruff as a code linter for the project

### Details and comments
After enabling ruff, this PR includes all the relevant fixes to make the ruff checker to pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
       - name: Run lint
         run: |
           python -m black --check .
+          python -m ruff check
```
Comment (jesus-talavera-ibm) [inline .github/workflows/ci.yml:34]: Are we going to keep both Black and Ruff?

File: qiskit_transpiler_service/__init__.py [hunk 2]
```diff
@@ -10,9 +10,9 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-from .ai import AIRouting
-from .transpiler_service import TranspilerService
-from .utils import create_random_linear_function, get_metrics
+from .ai import AIRouting  # noqa: F401
+from .transpiler_service import TranspilerService  # noqa: F401
+from .utils import create_random_linear_function, get_metrics  # noqa: F401
```
Comment (jesus-talavera-ibm) [inline qiskit_transpiler_service/__init__.py:15]: We could add this ignore `F401` to all `__init__.py` files
Here is an example: https://stackoverflow.com/questions/59167405/flake8-ignore-only-f401-rule-in-entire-file

Comment (cbjuan) [reply qiskit_transpiler_service/__init__.py]: You're right. Fixed _a la ruff_ in https://github.com/Qiskit/qiskit-ibm-transpiler/pull/48/commits/463fee3270d6deb9ec7c0710d4c9c367418e24b1

**Review by jesus-talavera-ibm — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jesus-talavera-ibm) [reply .github/workflows/ci.yml:34]: @Eric-Arellano confirmed that we can get rid of Black and use just Ruff

**Review by jesus-talavera-ibm — DISMISSED** [review]

**Review by jesus-talavera-ibm — APPROVED** [review]

**Comment by jesus-talavera-ibm** [comment]
@cbjuan This PR seems to have too many conflicts, since there've been a lot of changes since it was created. What do you think about closing this one and creating a fresh new PR?

**Comment by cbjuan** [comment]
Yes, let's close and open a new

**Pull request closed without merging**